packages/react-native-windows-init/src/Cli.ts

```diff
@@ -219,13 +217,12 @@ function getLatestMatchingVersion(
       // Ideally we'd be able to just use npm.packages.range(pkg, versionSemVer) here,
       // but alas it fails to give us the right result for react-native-windows@^0.60.0-0
       // as it fails to return pre-release versions
-      npm.packages.releases(
-        pkg,
-        (err: any, details: {[key: string]: Record<string, any>} | null) => {
-          if (err) {
-            reject(err);
-          } else if (details) {
-            const versions = Object.keys(details);
+      npmFetch.json(pkg, {registry: NPM_REGISTRY_URL}).then(
```

Comment:
It looks like this function previously wrapped a callback to return a promise. Because the API here already uses a promise, this might be clearer if we declare getLatestMatchingVersion as async (it already returns a promise) and use async/await.

E.g. `const response = await npmFetch.json(pkg, {registry: NPM_REGISTRY_URL})`.

Reply:
This also lets you use normal try/catch instead of manual rejection, error callbacks, etc.

Reply:
Converting this function to async makes it way simpler and easier to understand.